### ISSUE TITLE
Builds broken because package signature file entry is invalid

### ISSUE THREAD

**Issue by AArnott** [dotnet/sdk#2523]
As discussed [on this appveyor forum](https://help.appveyor.com/discussions/problems/16600-appveyor-suddenly-totally-broken), AppVeyor builds recently started breaking for me.  Feodor Fitsner kindly came up with a minimal repro that suggests that the .NET Core 2.1 packages on nuget.org have bad signatures.

**Repro**

Create a test-nuget.csproj file with these contents:

```xml
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFrameworks>netcoreapp1.0;netcoreapp2.0;netcoreapp2.1</TargetFrameworks>
  </PropertyGroup>
</Project>
```

Rename or remove your `%userprofile%\.nuget\packages` folder.

Restore packages for `test-nuget.csproj` *using msbuild*:

```ps1
msbuild /t:restore /p:RestoreDisableParallel=true,TreatWarningsAsErrors=true
```

This produces these errors:

```
  Restoring packages for D:\temp\test-nuget\test-nuget.csproj...
<snip/>
  Installing Microsoft.NETCore.DotNetAppHost 2.0.0.
  Installing Microsoft.NETCore.App 2.1.0.
D:\temp\test-nuget\test-nuget.csproj : error NU3005: The package signature file entry is invalid. The central directory header field 'compression method' has an in
valid value (8).
  Installing Microsoft.NETCore.DotNetHostPolicy 2.1.0.
D:\temp\test-nuget\test-nuget.csproj : error NU3005: The package signature file entry is invalid. The central directory header field 'compression method' has an in
valid value (8).
  Installing Microsoft.NETCore.Platforms 2.1.0.
D:\temp\test-nuget\test-nuget.csproj : error NU3005: The package signature file entry is invalid. The central directory header field 'compression method' has an in
valid value (8).
  Installing Microsoft.NETCore.Targets 2.1.0.
D:\temp\test-nuget\test-nuget.csproj : error NU3005: The package signature file entry is invalid. The central directory header field 'compression method' has an in
valid value (8).
  Installing NETStandard.Library 2.0.3.
  Installing Microsoft.NETCore.DotNetHostResolver 2.1.0.
D:\temp\test-nuget\test-nuget.csproj : error NU3005: The package signature file entry is invalid. The central directory header field 'compression method' has an in
valid value (8).
  Installing Microsoft.NETCore.DotNetAppHost 2.1.0.
D:\temp\test-nuget\test-nuget.csproj : error NU3005: The package signature file entry is invalid. The central directory header field 'compression method' has an in
valid value (8).
  Generating MSBuild file D:\temp\test-nuget\obj\test-nuget.csproj.nuget.g.props.
  Generating MSBuild file D:\temp\test-nuget\obj\test-nuget.csproj.nuget.g.targets.
  Restore failed in 32.27 sec for D:\temp\test-nuget\test-nuget.csproj.
```

Interestingly, this only repros while there are multiple target frameworks specified. Reducing it to just `netcoreapp2.1` eliminates the error.

**Comment by nkolev92** [comment]
Did some digging cause this looks weird, if you do a NuGet.exe verify -signatures on the package, for example Microsoft.NETCore.App 2.1.0, you get the same error message. 

This seems to suggest that there is something wrong with this package. 

The reason why it fails with multitargetting is because of the fallback folders.
 When you cross-target 2.x and 1.x, the new fallback folder from `C:\Program Files\dotnet\sdk\NuGetFallbackFolder` is not used. 

When you target 2.x only, then the package is consumed from the fallback folder, instead of downloading it and installing in the global packages folder. 

tl;dr; 

- The multi-targeting scenario is by design because NuGet needs to download that package. 
- Rest of it suggests that the package is not signed correctly.

@dtivel @PatoBeltran will be able to give more info on the 2nd one.

**Comment by mmitche** [comment]
This is really odd. Why would it start now? 2.1.0 is ages old.  Was there a change in nuget?

**Comment by mmitche** [comment]
WIth nuget.exe 4.6.2.5055 I get a successful verification

**Comment by jainaashish** [comment]
We also run an additional exercise to validate packages hash from extracted SHA512 file vs generating SHA512 hash from nupkg in dotnet fallback folder and all these package validation failed which suggest either dotnet is not extracting nupkg correctly in fallback folder or they are creating a new nupkg in fallback folder. 

The right expectation here is that dotnet has single nupkg for each of their package, which is extracted into fallback folder using NuGet and keep the same nupkg there or publish the same nupkg to NuGet.org or any other feed. At no point in time, there should be another nupkg generated for the same package (on the fly or anything) after the original nupkg has been extracted in fallback folder.

**Comment by AArnott** [comment]
What's happening here? Now I'm seeing this error in VS itself, blocking intellisense and builds:

> Error occurred while restoring NuGet packages: The package signature file entry is invalid. The central directory header field 'compression method' has an invalid value (8).

**Comment by livarcocc** [comment]
I am going to move this issue over to NuGet, where we can get quicker and better answers to this.

**Comment by livarcocc** [comment]
This issue was moved to NuGet/Home#7414
